src/python/pants/goal/run_tracker.py

```diff
@@ -12,6 +12,7 @@
 import uuid
 from collections import OrderedDict
 from contextlib import contextmanager
+from dataclasses import dataclass, field
```

Comment:
Nit: `field` is a really general name, I recommend using `import dataclasses` so that you can use `dataclasses.field`. It's fine to still use a bare `@dataclass`.